server/grouping/grouping.go

```diff
@@ -138,12 +138,17 @@ func isInGroup(matchStatements []map[string]string, factmap map[string][]string)
 				continue
 			}
 
-			regexStrs := factmap["Hostname"]
-			for j := range matchStatements {
-				result := exp.Find([]byte(regexStrs[j]))
-				if result != nil {
-					return true
-				}
+			// retrieved hostname from system
+			//
+			// TODO(mierdin): Currently, hostname is provided as single
+			// string in slice, so we can go straight to zero index.
+			// This is due to crappy implementation in the fact collecting
+			// process, so this is an acceptable temporary workaround.
+			hostname := factmap["Hostname"][0]
```

Comment:
Is it possible for the hostname to not exist in the map or for the slice to be empty?

Reply:
It is, but this code won't run due to `if _, ok := factmap["Hostname"]; !ok {` a few lines up